src/ArchimedeanCopula.jl

```diff
@@ -198,3 +187,101 @@ function Distributions._rand!(rng::Distributions.AbstractRNG, ::ArchimedeanCopul
     return A
 end
 
+function Distributions._rand!(rng::Distributions.AbstractRNG, C::ClaytonCopula, x::AbstractVector{<:Real})
+    x[:] = inverse_rosenblatt(C, rand(rng, length(x)))
+    return x
+end
+function Distributions._rand!(rng::Distributions.AbstractRNG, C::ClaytonCopula, A::DenseMatrix{<:Real})
+    A[:] = inverse_rosenblatt(C, rand(rng, size(A)...))
+    return A
+end
+
+function rosenblatt(C::ArchimedeanCopula{d,TG}, u::AbstractMatrix{<:Real}) where {d,TG}
+    @assert d == size(u, 1)
+
+    U = zeros(eltype(u), size(u))
+    U[1, :] = u[1, :]
+
+    for j in 2:d
+        U[j, :] .=
+            ϕ⁽ᵏ⁾.(C.G, Val(j - 1), reduce(+, [ϕ⁻¹.(C.G, u[k, :]) for k in 1:j])) ./ ϕ⁽ᵏ⁾.(C.G, Val(j - 1), reduce(+, [ϕ⁻¹.(C.G, u[k, :]) for k in 1:(j - 1)]))
+    end
+
+    return U
+end
+
+function rosenblatt(
+    C::ArchimedeanCopula{d,TG}, u::AbstractMatrix{<:Real}
+) where {d,TG<:ClaytonGenerator}
+    @assert d == size(u, 1)
+
+    U = zeros(eltype(u), size(u))
+    U[1, :] = u[1, :]
+
+    for j in 2:d
+        U[j, :] .=
+            (
+                (1 .- j .+ sum(u[1:j, :] .^ (-C.G.θ); dims=1)[:]) ./
+                (2 .- j .+ sum(u[1:(j - 1), :] .^ (-C.G.θ); dims=1)[:])
+            ) .^ (-1 / C.G.θ - (j - 1))
+    end
+    return U
+end
+
+function rosenblatt(C::ArchimedeanCopula{d,TG}, u::AbstractVector{<:Real}) where {d,TG}
+    @assert d == size(u, 1)
+
+    return rosenblatt(C, reshape(u, (d, 1)))[:]
+end
+
+function inverse_rosenblatt(
+    C::ArchimedeanCopula{d,TG}, u::AbstractMatrix{<:Real}
+) where {d,TG}
+    @assert d == size(u, 1)
+
+    U = zeros(eltype(u), size(u))
+    U[1, :] = u[1, :]
+
+    for i in axes(u, 2)
+        for j in 2:d
+            f =
+                x ->
+                    (
+                        ϕ⁽ᵏ⁾(C.G, Val(j - 1), sum(ϕ⁻¹.(C.G, [U[1:(j - 1), i]..., x]))) /
+                        ϕ⁽ᵏ⁾(C.G, Val(j - 1), sum(ϕ⁻¹.(C.G, U[1:(j - 1), i])))

```

Comment:
Here too there is a vector `[U[1:(j - 1), i]..., x]``that could simply be replaced by a tuple `()` or even you could replace `sum(ϕ⁻¹.(C.G, [U[1:(j - 1), i]..., x])))` by `sum(ϕ⁻¹.(C.G, U[1:(j - 1), i]))+ϕ⁻¹(x)` for cleanliness

Reply:
In fact yes, the `sum(ϕ⁻¹.(C.G,...)))` could be precomputed before instantiating the function to reduce its cost drastically.